src/realm/table.hpp

```diff
@@ -1161,6 +1161,20 @@ class Table {
     const BacklinkColumn& get_column_backlink(size_t ndx) const noexcept;
     BacklinkColumn& get_column_backlink(size_t ndx);
 
+    void verify_column(size_t col_ndx, const ColumnBase* col) const
+    {
+        if (col_ndx < m_cols.size()) {
+            if (m_cols[col_ndx] == col)
+                return;
+            // A column could have been inserted. Check if col can be found further on
+            for (auto it = m_cols.begin() + col_ndx + 1; it < m_cols.end(); ++it) {
+                if (*it == col)
+                    return;
+            }
+        }
+        throw LogicError(LogicError::column_index_out_of_range);
```

Comment:
This seems fine to me, but you might want to confirm with @simonask to see if he wants a unique exception to be thrown here, to differentiate this scenario from other `column_index_out_of_range` errors.